modules/misc/meta.nix

```diff
@@ -0,0 +1,73 @@
+{ lib, ... }:
+# Based on https://github.com/NixOS/nixpkgs/blob/814a4e48/nixos/modules/misc/meta.nix
+let
+  maintainer = lib.mkOptionType {
+    name = "maintainer";
+    check = m: lib.elem m (lib.attrValues lib.maintainers);
+    merge =
+      loc: defs:
+      let
+        def = lib.last defs;
+      in
+      {
+        ${def.file} = def.value;
+      };
+  };
+
+  listOfMaintainers = lib.types.listOf maintainer // {
+    # Returns attrset of
+    #   { "module-file" = [
+    #        "maintainer1 <first@nixos.org>"
+    #        "maintainer2 <second@nixos.org>" ];
+    #   }
+    merge =
+      loc: defs:
+      lib.pipe defs [
+        (lib.imap1 (
+          n: def:
+          lib.imap1 (
+            m: value:
+            maintainer.merge (loc ++ [ "[${toString n}-${toString m}]" ]) [
+              {
+                inherit (def) file;
+                inherit value;
+              }
+            ]
+          ) def.value
+        ))
+        lib.flatten
+        lib.zipAttrs
+      ];
+  };
+in
+{
+  options.meta = {
+    maintainers = lib.mkOption {
+      type = listOfMaintainers;
+      internal = true;
+      default = [ ];
+      example = lib.literalExpression "[ lib.maintainers.all ]";
+      description = ''
+        List of maintainers of each module.  This option should be defined at
+        most once per module.
+      '';
+    };
+
+    # TODO: we could do this instead of nixvimInfo.description
+    # doc = lib.mkOption {
+    #   type = lib.types.path // {
+    #     # Returns tuples of
+    #     # { file = "module location"; value = <path/to/doc.xml>; }
+    #     merge = loc: defs: defs;
+    #   };
+    #   internal = true;
+    #   example = "./meta.chapter.md";
+    #   description = ''
+    #     Documentation prologue for the set of options of each module.  This
+    #     option should be defined at most once per module.
+    #   '';
+    # };
```

Comment:
Yeah I agree that we should maybe at some point merge our nixvimInfo with this module, as they basically serve the same purpose. Though let's do that later

Reply:
Shall we keep the commented code as a reminder, or remove it and open an issue?

Reply:
I think an issue is a better reminder

Reply:
Opened https://github.com/nix-community/nixvim/issues/2309